### PR TITLE
update menu on hash change

### PR DESCRIPTION
**Pull request by qlj-lijuan**
Realize the function of update the menu when the url changes.

### CODE REVIEW

**Comment by qlj-lijuan** [comment]
> Perhaps it's time to get multiple dev pages - one for charts, and one for the pv-tree. Could you move out the menu and search box to a new dev page?

Sure, created a new mage menu_dev for pv-tree and search box

File: server/templates/menu_dev.html [hunk 1]
```diff
@@ -0,0 +1,47 @@
+<!--
```
Comment (beets) [inline server/templates/menu_dev.html:1]: Nit: Please use jinja comments {# ... #}
That won't render with the page load.

File: server/templates/dev.html [hunk 2]
```diff
@@ -21,6 +21,9 @@
   <link rel="stylesheet" href={{url_for('static', filename='css/draw.css')}}></script>
   <link rel="stylesheet" href={{url_for('static', filename='css/dev.css')}}></script>
   <link rel="stylesheet" href={{url_for('static', filename='css/gni.css')}}></script>
+  <link rel="stylesheet" href={{url_for('static', filename='css/place_overview_search.css')}}></script>
```
Comment (shifucun) [inline server/templates/dev.html:24]: these can be removed?

File: static/js/pv_tree.tsx [hunk 3]
```diff
@@ -150,10 +188,38 @@ class Menu extends Component<MenuPropType, {}> {
   }
 }
 
-class page extends Component<MenuPropType, {}> {
+interface pageStateType {
+  statvarPaths: string[][];
+}
+
+class page extends Component<MenuPropType, pageStateType> {
```
Comment (shifucun) [inline static/js/pv_tree.tsx:195]: Use "Page" as class name

File: server/templates/menu_dev.html (hunk 1 above)
Comment (qlj-lijuan) [reply server/templates/menu_dev.html]: got it.

File: server/templates/dev.html (hunk 2 above)
Comment (qlj-lijuan) [reply server/templates/dev.html]: removed

File: static/js/pv_tree.tsx (hunk 3 above)
Comment (qlj-lijuan) [reply static/js/pv_tree.tsx]: Done

**Pull request merged**